src/java.base/share/classes/java/lang/reflect/Executable.java

```diff
@@ -337,17 +345,36 @@ Type[] getAllGenericParameterTypes() {
             if (realParamData) {
                 final Type[] out = new Type[nonGenericParamTypes.length];
                 final Parameter[] params = getParameters();
-                int fromidx = 0;
-                for (int i = 0; i < out.length; i++) {
-                    final Parameter param = params[i];
-                    if (param.isSynthetic() || param.isImplicit()) {
-                        // If we hit a synthetic or mandated parameter,
-                        // use the non generic parameter info.
-                        out[i] = nonGenericParamTypes[i];
-                    } else {
-                        // Otherwise, use the generic parameter info.
-                        out[i] = genericParamTypes[fromidx];
-                        fromidx++;
+                if (getDeclaringClass().isRecord() && this instanceof Constructor) {
+                    /* we could be seeing a compact constructor of a record class
+                     * its parameters are mandated but we should be able to retrieve
+                     * its generic information if present
+                     */
+                    int fromidx = genericParamTypes.length - 1;
+                    for (int i = out.length - 1; i >= 0; i--) {
```

Comment:
We don't really need this loop any more; can just do something like:
```java
if (genericParamTypes.length == nonGenericParamTypes.length) {
    out = genericParamTypes;
} else {
    out = nonGenericParamTypes.clone();
}
```

Reply:
yup I like this, thanks